api_tester/lib/purchases_flutter_api_test.dart

```diff
@@ -0,0 +1,420 @@
+import 'package:purchases_flutter/purchases_flutter.dart';
+
+// ignore_for_file: unused_element
+// ignore_for_file: unused_local_variable
+// ignore_for_file: deprecated_member_use
+class _PurchasesFlutterApiTest {
+
+  void _checkSetup() {
+    String apiKey = "fakeApiKey";
+    String? userId = "fakeUserId";
+    bool observerMode = false;
+    String? userDefaultsSuiteName = "fakeSuiteName";
+    bool useAmazon = false;
+    Future<void> callback = Purchases.setup(
+      apiKey,
+      appUserId: userId,
+      observerMode: observerMode,
+      userDefaultsSuiteName: userDefaultsSuiteName,
+      useAmazon: useAmazon
+    );
+  }
+
+  void _checkConfigure() {
+    PurchasesConfiguration configuration = PurchasesConfiguration("fakeApiKey");
+    Future<void> callback = Purchases.configure(configuration);
+  }
+
+  void _checkSetFinishTransactions() {
+    bool finishTransactions = false;
+    Future<void> callback = Purchases.setFinishTransactions(finishTransactions);
+  }
+
+  void _checkSetAllowSharingStoreAccount() {
+    bool allowSharing = false;
+    Future<void> callback = Purchases.setAllowSharingStoreAccount(allowSharing);
+  }
+
+  void _checkAddCustomerInfoUpdateListener() {
+    void customerInfolistener(CustomerInfo customerInfo) {}
+    Purchases.addCustomerInfoUpdateListener(customerInfolistener);
+  }
+
+  void _checkRemoveCustomerInfoUpdateListener() {
+    void customerInfolistener(CustomerInfo customerInfo) {}
+    Purchases.removeCustomerInfoUpdateListener(customerInfolistener);
+  }
+
+  void _checkAddReadyForPromotedProductPurchaseListener() {
+    void listener(
+        String productIdentifier,
+        Future<PromotedPurchaseResult> Function() startPurchase
+        ) {}
+    Purchases.addReadyForPromotedProductPurchaseListener(listener);
+  }
+
+  void _checkRemoveReadyForPromotedProductPurchaseListener() {
+    void listener(
+        String productIdentifier,
+        Future<PromotedPurchaseResult> Function() startPurchase
+        ) {}
+    Purchases.removeReadyForPromotedProductPurchaseListener(listener);
+  }
+
+  void _checkGetOfferings() async {
```

Comment:
Note how I'm marking some of these methods as `async`. This is so we can test `async` methods easily. Right now we have a mix, with some allowing `async` and some not.

On these tests, I'm testing both the `Future` value returned, and the `await` system, to make sure there are no regressions on that part. I thought we could only test the `await` system... But I wasn't sure, so I added both. Lmk if you think only with the `await` system would be enough.